src/ol_openedx_course_structure_api/views.py

```diff
@@ -0,0 +1,96 @@
+"""Views for Course structure API"""
+
+import logging
+
+from lms.djangoapps.courseware.management.commands.dump_course_structure import (
+    dump_block,
+)
+from opaque_keys import InvalidKeyError
+from opaque_keys.edx.keys import CourseKey
+from openedx.core.lib.api.view_utils import (
+    DeveloperErrorViewMixin,
+    verify_course_exists,
+)
+from rest_framework import status
+from rest_framework.generics import GenericAPIView
+from rest_framework.permissions import IsAdminUser
+from rest_framework.response import Response
+from xmodule.modulestore.django import modulestore
+from xmodule.modulestore.inheritance import compute_inherited_metadata
+
+log = logging.getLogger(__name__)
+
+
+class CourseStructureView(DeveloperErrorViewMixin, GenericAPIView):
+    """
+    An API View for course structure
+
+    **Example Requests**
+
+    GET api/course-structure/v0/{course_id}/
+    GET api/course-structure/v0/{course_id}/?inherited_metadata=true&inherited_metadata_default=true
+
+    **Example GET Response**
+
+    The API will return 200, 404 response codes
+
+    A 404 would be returned if course ID is invalid or course not found
+    A 200 response would look something like
+    {
+      "$block_url": {
+        "category": "$block_category",
+        "children": [$block_children_urls... ],
+        "metadata": {$block_metadata}
+      },
+      "$block_url": ....
+    }
+
+
+    """  # noqa: E501
+
+    http_method_names = ["get"]
+    permission_classes = [
+        IsAdminUser,
+    ]
+
+    @verify_course_exists()
+    def get(self, request, course_id):
```

Comment:
@pdpinch do we want this endpoint to require authentication? I would presume that this should only be allowed for admin users so that we don't leak information about unreleased course elements to students.

Reply:
oh, this endpoint requires Admin, which is on line 52.